Feat/signup login

dicohy27 opened the pull request:
로그인, 회원가입 기능 구현 완료

dicohy27 committed:
음식 검색 -> 선택 -> 양 입력 하면 DietEditPage에 음식이름과 양 추가 기능

dicohy27 committed:
OneSetRecordWidget TextField 텍스트 위치 가운데 정렬

dicohy27 committed:
버전관련?

dicohy27 committed:
Material3으로 했더니 버튼 글자 안보여서 색 바꿈

dicohy27 committed:
FlutterSecureStorage, Dio, DartJsonwebtoken 패키지 설치

dicohy27 committed:
readme.md

dicohy27 committed:
지가 알아서 바뀜

dicohy27 committed:
login, signup api 주소

dicohy27 committed:
AuthService 클래스 구현

dicohy27 committed:
Provider에 AuthService 추가

dicohy27 committed:
로그인 기능 구현

dicohy27 committed:
로그아웃 구현

dicohy27 committed:
MenuRoutingPage의 Action을 PopupMenuButton으로 변경

dicohy27 committed:
API클래스 수정

dicohy27 committed:
로그인, 회원가입 페이지에서 화면 아무데나 누르면 키보드 내려가게 함

dicohy27 committed:
SignupPage 불필요한 함수 제거, 회원가입 버튼 AuthService와 연결, validate를 AuthService로 넘김

dicohy27 committed:
http -> dio로 변경

dicohy27 committed:
회원가입 성공 시 로그인 페이지로 이동 안되는 문제 해결 : Get.back(closeOverlays: true)

dicohy27 commented on lib/api/api.dart:
```diff
@@ -1,7 +1,10 @@
-  static const hostConnect = "http://192.168.0.41/api_new_members";
+  static const hostConnect =
+      "http://fitty-develop-api.ap-northeast-2.elasticbeanstalk.com";
   static const hostConnectUser = "$hostConnect/user";
 
   static const signup = "$hostConnect/user/signup.php";
   static const validateEmail = "$hostConnect/user/validate_email.php";
```
얘넨 이제 안쓸듯

uuccu replied:
안물안궁

dicohy27 replied:
아가리해

dicohy27 commented on lib/page/login_signup/login_page.dart:
```diff
@@ -16,100 +18,130 @@ class _LoginPageState extends State<LoginPage> {
+
+  _asyncMethod() async {
+    userInfo = await storage.read(key: "userInfo");
+    // todo : userInfo 검증 필요
```
@uuccu 지금은 그냥 userInfo가 존재하면 바로 로그인 페이지 나가게(MenuRoutingPage로 이동)되어있는데 이걸 이제 jwt토큰을 서버에 보내서 검증 받는 식으로 해야되냐?

uuccu replied:
음 그거보다 바로 MenuRoutingPage로 들어가는데에 필요한 API에서 토큰 verifying 에러가 발생하면 로그인페이지로 가게 하는게 나을듯

dicohy27 replied:
토큰 verifying 에러는 서버에서 보내줌?

uuccu replied:
ㅇㅇ

dicohy27 committed:
fix : 스크롤 내렸을 때 음식들 가끔 못불러오던 문제 해결
이전에 검색했던 음식의 목록을 전부 불러왔을 때 hasNextPage가 false로 업데이트 된 뒤 다른 음식 검색했을때도 여전히 값이 false로 남아서 생긴 문제

dicohy27 committed:
음식 양을 선택 안하고 뒤로가기 했을 때 null 체크

dicohy27 committed:
dart fix

dicohy27 committed:
유저 정보 있으면 LoginPage거치지 않고 바로 MenuRoutingPage로 가게 함

dicohy27 committed:
FlutterNativeSplash 패키지 설치

uuccu commented:
스샷왜 안올리냐

uuccu commented:
아니 움짤

dicohy27 commented:
옹량맞추기 짜증남

dicohy27 committed:
인트로 페이지인데 아직 안써서 비워둠

dicohy27 committed:
main에서 실행되는 비동기 함수 FutureBuilder로 이동
처음 플러터 마크 있는 인트로 화면에서 오래 머물길래 gif있는 로딩화면에서 실행되게 해서 플러터 마크 인트로 화면 빨리 넘어가게 하려 했는데 별 차이 없음 뻘짓인듯.

uuccu commented on README.md:
```diff
@@ -1 +1,5 @@
 # Fitti-Frontend-App
+
+asset/config/ 안에 .env 파일 만들고 FOOK_API_KEY=키 값 추가해야 함
+![image](https://github.com/FITTY-APP/Fitti-Frontend-App/assets/59111693/6e6e7107-dd9e-4c1f-8a6b-7f8174045979)
+
```
??

dicohy27 replied:
뭐말하는지모를까봐 이미지 첨부함

uuccu replied:
FOOK_API_KEY가 뭐임

dicohy27 replied:
디코에 올렸잖음 고정메세지로

uuccu replied:
그게 뭐하는 키인지 물어본거였음

uuccu commented on lib/api/api.dart:
```diff
@@ -1,7 +1,10 @@
 class API {
-  static const hostConnect = "http://192.168.0.41/api_new_members";
+  static const hostConnect =
+      "http://fitty-develop-api.ap-northeast-2.elasticbeanstalk.com";
   static const hostConnectUser = "$hostConnect/user";
 
   static const signup = "$hostConnect/user/signup.php";
   static const validateEmail = "$hostConnect/user/validate_email.php";
+  static const login = "$hostConnect/auth/login";
+  static const register = "$hostConnect/auth/register";
 }
```
음.... //todo profile이라고 주석처리점

dicohy27 replied:
함

uuccu commented on lib/data/auth_service.dart:
```diff
@@ -0,0 +1,105 @@
+
+class AuthService extends ChangeNotifier {
+  User? currentUser;
+
+  void setUser(User? user) {
+    currentUser = user;
+    notifyListeners();
+  }
+
+  void removeUser() {
+    currentUser = null;
+    notifyListeners();
+  }
+
```
Listner쓴거같은데 어떨때를 위한애임?

dicohy27 replied:
쟤 쓰면 Consumer로 감싼 애들 ui가 리로드됨

uuccu commented on lib/data/auth_service.dart:
```diff
@@ -0,0 +1,105 @@
+      onFail("이름을 입력해주세요.");
+      return;
+    }
+    try {
+      var dio = Dio();
+      Response response = await dio.post(API.register, data: {
+        "email": email,
+        "password": password,
+        "name": name,
+      });
+      if (response.statusCode == 200) {
+        onSuccess();
+        notifyListeners();
+      } else {
+        onFail("회원가입에 실패했습니다.");
+      }
+    } catch (e) {
+      onFail("회원가입에 실패했습니다.");
+    }
+  }
```
try catch문을 왜 이렇게 작성한거임?

dicohy27 replied:
어디가 이상한거임? 코파일럿이 만들어준거 약간 수정한건데

uuccu replied:
아 어쩐지 조금 이상하더라
 try catch문 안에서 onFAIL(애가 통상적인 예외 발생 throw랑 동일하게 생각하고 말하는거임)을 쓸거면 catch에다가만 쓰는게 일반적이니까 한말이었음. try안에 onFail이 또 있으니까. 애초에 try에서 실패하면 catch로 갈탠데 try catch 쓰는의미가 좀 퇴색되는느낌.

dicohy27 replied:
try에서 실패 안했는데 회원가입이 실패하는 경우(response.statusCode가 200이 아닐 경우) try문에서 onFail 실햄됨

dicohy27 replied:
try에서 실패 안해도 회원가입 실패할 수 있잖음

uuccu replied:
ㅇㅇ 논리상 이상하다는게 아니고 try catch문을 개떡같이 사용했다는 의미임

uuccu replied:
걍 넘기삼

uuccu commented on lib/data/auth_service.dart:
```diff
@@ -0,0 +1,105 @@
+
+    try {
+      var dio = Dio();
+      Response response = await dio.post(API.login, data: {
+        "email": email,
+        "password": password,
+      });
+      if (response.statusCode == 200) {
```
이렇게 api 호출하는군요

uuccu replied:
근데 왜 dio라는 객체를 통해서 접근하는거임?

dicohy27 replied:
ㅇㅇ

uuccu replied:
왜 대답안하고 resolve하냐

dicohy27 replied:
dio 질문 못봄 dio가 http 상위호환 패키지라 씀

uuccu replied:
dio가 패키지임? 패키지 링크점

dicohy27 replied:
https://pub.dev/packages/dio

uuccu commented on lib/main.dart:
```diff
@@ -1,41 +1,56 @@
-          }),
+        future: () async {
+          var myExerciseRecord = context.read<MyExerciseRecord>();
+          await myExerciseRecord.initDateTimeToDayExerciseRecordMap();
+          await initializeDateFormatting();
+          await dotenv.load(fileName: 'asset/config/.env');
+
+          var userInfo = await storage.read(key: "userInfo");
+          if (userInfo != null) {
+            return userInfo;
+          }
```
key을 이거 하나로만 저장할거면 상수로쓰고. key값을 겹치지 않게 해야할듯. 앱깔고 계정 2개로 로그인왔다갔다가 안된다는거니까

dicohy27 replied:
계정 2개 자동 로그인 정보를 저장할 필요는 없지 않음? 하나 로그아웃 하고 다른 계정으로 로그인 하려면 어차피 이메일 비번 입력해야되니까

uuccu replied:
ㅇㅇ 필요없을듯. 상수로 하삼

dicohy27 replied:
완료

uuccu commented on lib/page/diet/food_amount_input_page.dart:
```diff
@@ -0,0 +1,41 @@
+
+class FoodAmountInputPage extends StatelessWidget {
+  FoodAmountInputPage({super.key, required this.foodName});
+
+  final String foodName;
+  final TextEditingController foodAmountController = TextEditingController();
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(foodName),
+      ),
+      body: Center(
+        child: Column(
+          children: [
+            TextField(
+              controller: foodAmountController,
```
? 로그인만 만든게 아니네 ㅅㅂ 어쩐지 이상하더라

dicohy27 replied:
뭐가

uuccu replied:
식단은 분리해야지

dicohy27 replied:
ㅇㅇ

uuccu commented on lib/page/menu_routing_page.dart:
```diff
@@ -38,12 +44,25 @@ class _MenuRoutingPageState extends State<MenuRoutingPage> {
+                  for (final value in MenuType.values)
+                    PopupMenuItem(
+                      value: value,
+                      child: Text(value.toString()),
+                      onTap: () {
+                        switch (value) {
+                          case MenuType.logOut:
+                            storage.delete(key: "userInfo");
+                            Get.off(() => const LoginPage());
+                            break;
+                          case MenuType.setting:
+                            break;
+                        }
```
logout 버튼도 만들었냐? 굳굳 대단하네.

dicohy27 replied:
ㅇㅇ

uuccu reviewed (changes requested):
고생했음. 
근데 storage에 저장하는 코드는 어디에 있냐? 보이지가 않는데

dicohy27 committed:
자동생성

dicohy27 committed:
todo profile 주석처리

dicohy27 commented:
> 고생했음. 근데 storage에 저장하는 코드는 어디에 있냐? 보이지가 않는데

무슨 storage 말하는거임 운동정보 아님 로그인정보?

uuccu commented:
> 

로그인 토큰 저장하는 코드가 안보여서 물어본거임

dicohy27 commented on lib/data/auth_service.dart:
```diff
@@ -0,0 +1,105 @@
+        Map<String, dynamic> payload = jwt.payload;
+        String email = payload["email"];
+        var val = jsonEncode({"email": email, "token": token});
+        onSuccess(val);
```
여기가 로그인 정보 저장하는 부분

uuccu replied:
onSuccess가 저장임?

dicohy27 replied:
ㅇㅇ

dicohy27 commented on lib/page/login_signup/login_page.dart:
```diff
@@ -16,100 +18,112 @@ class _LoginPageState extends State<LoginPage> {
+                          onFail: (err) {
+                            Get.snackbar("로그인 실패", err);
+                          },
+                          onSuccess: (val) async {
```
얘가 AuthService 에서의 onSuccess와 같음

dicohy27 reviewed (commented):
로그인 정보 저장

dicohy27 committed:
userInfo 상수 처리(.env에 저장)
